### DISCUSSION TITLE
Is it possible to hide some shapes in a loaded scene?

### DISCUSSION THREAD

**Discussion by wylighting** [mitsuba-renderer/mitsuba3#681] [Q&A] [answered]
Hi~

I want to randomly remove (hide) some shapes in a scene to render many image samples, but I am not sure the best practice to achieve this. One possible way is to modify the scene description and use 'mi.load_dict' for each data sample, but it will cause the duplicated loading process which is slow. So, it is possible to achieve this in a loaded scene directly? For example, can I specify a shape to be excluded in the ray intersection process?

**Comment by njroussel** [comment, marked as answer]
Hi @swordigo1995 

Currently there is no support for something like this. There are two possible workarounds though:

1. Use the [`mask` BSDF](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#opacity-mask-mask) to make the shape transparent. It is very likely that this will producer noisier images as shadow rays will still intersect the shape.
2. Move the shapes far away from your scene by updating their `to_world` parameter. This would re-build acceleration structure for the ray-shape intersections, depending on your scene this shouldn't bee too slow.

**Reply by wylighting** [reply]
Thanks!
